### PR TITLE
[SPARK-19460][SparkR]:Update dataset used in R documentation, examples to reduce warning noise and confusions

### PR DESCRIPTION
**Pull request by wangmiao1981**
## What changes were proposed in this pull request?

Replace `iris` dataset with `Titanic` or other dataset in example and document.

## How was this patch tested?

Manual and existing test

### CODE REVIEW

**Comment by wangmiao1981** [comment]
cc @felixcheung

**Review by srowen — APPROVED** [review]

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 1]
```diff
@@ -565,11 +565,10 @@ We use a simple example to demonstrate `spark.logit` usage. In general, there ar
...
 Binomial logistic regression
-```{r, warning=FALSE}
-df <- createDataFrame(iris)
-# Create a DataFrame containing two classes
-training <- df[df$Species %in% c("versicolor", "virginica"), ]
-model <- spark.logit(training, Species ~ ., regParam = 0.00042)
+```{r}
+t <- as.data.frame(Titanic)
+training <- createDataFrame(t)
+model <- spark.logit(training, Sex ~ ., regParam = 0.00042)
```
Comment (felixcheung) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: would be good to check if the regParam value make sense in the generated output?

File: examples/src/main/r/ml/bisectingKmeans.R [hunk 2]
```diff
@@ -25,20 +25,21 @@ library(SparkR)
...
 # Fit bisecting k-means model with four centers
-model <- spark.bisectingKmeans(df, Sepal_Length ~ Sepal_Width, k = 4)
+model <- spark.bisectingKmeans(training, Class ~ Survived, k = 4)
 
 # get fitted result from a bisecting k-means model
 fitted.model <- fitted(model, "centers")
 
 # Model summary
-summary(fitted.model)
+head(summary(fitted.model))
```
Comment (felixcheung) [inline examples/src/main/r/ml/bisectingKmeans.R]: summary should print without having to do a `head` here?

File: examples/src/main/r/ml/glm.R [hunk 3]
```diff
@@ -25,11 +25,12 @@ library(SparkR)
...
 # $example on$
-irisDF <- suppressWarnings(createDataFrame(iris))
+t <- as.data.frame(Titanic)
+training <- createDataFrame(t)
 # Fit a generalized linear model of family "gaussian" with spark.glm
-gaussianDF <- irisDF
-gaussianTestDF <- irisDF
-gaussianGLM <- spark.glm(gaussianDF, Sepal_Length ~ Sepal_Width + Species, family = "gaussian")
+gaussianDF <- training
+gaussianTestDF <- training
```
Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: I think this example is a bit weird - it takes the same data to build the model and then predict with it.
I suspect we are really limited in terms of how much data we have here, but we should consider building a better example which include doing a randomSplit into training and test set etc..

Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: ditto for binomial here or kmeans.R, ml.R

File: examples/src/main/r/ml/bisectingKmeans.R (hunk 2 above)
Comment (wangmiao1981) [inline examples/src/main/r/ml/bisectingKmeans.R]: In this case, `summary` returns a DataFrame. It won't print out the contents of the DataFrame.

File: R/pkg/vignettes/sparkr-vignettes.Rmd (hunk 1 above)
Comment (wangmiao1981) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: I will check.

File: R/pkg/R/mllib_tree.R [hunk 4]
```diff
@@ -143,14 +143,15 @@ print.summary.treeEnsemble <- function(x) {
...
+#' df <- createDataFrame(t)
+#' model <- spark.gbt(df, Sex ~ Age + Freq, "classification")
 #'
 #' # numeric label is also supported
-#' iris2 <- iris[iris$Species != "virginica", ]
-#' iris2$NumericSpecies <- ifelse(iris2$Species == "setosa", 0, 1)
-#' df <- createDataFrame(iris2)
-#' model <- spark.gbt(df, NumericSpecies ~ ., type = "classification")
+#' t2 <- as.data.frame(Titanic)
+#' t2$NumericGener <- ifelse(t2$Sex == "Male", 0, 1)
```
Comment (wangmiao1981) [inline R/pkg/R/mllib_tree.R]: `Gender`

File: R/pkg/R/mllib_tree.R [hunk 5]
```diff
@@ -143,14 +143,15 @@ print.summary.treeEnsemble <- function(x) {
 #'
 #' # fit a Gradient Boosted Tree Classification Model
 #' # label must be binary - Only binary classification is supported for GBT.
-#' df <- createDataFrame(iris[iris$Species != "virginica", ])
-#' model <- spark.gbt(df, Species ~ Petal_Length + Petal_Width, "classification")
+#' t <- as.data.frame(Titanic)
+#' df <- createDataFrame(t)
+#' model <- spark.gbt(df, Sex ~ Age + Freq, "classification")
```
Comment (wangmiao1981) [inline R/pkg/R/mllib_tree.R]: Just want to demonstrate with a category variable. I can change it to `Survived`. Is it ok?

File: examples/src/main/r/ml/glm.R (hunk 3 above)
Comment (wangmiao1981) [inline examples/src/main/r/ml/glm.R]: Yes. I agree. I saw other examples using the same dataset as testing. How about fixing them all in another follow-up PR? We only focus on fixing the `iris` dataset replacement in this PR. Thanks!

Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: we could but as I've mentioned, Titanic is really small - it might not work properly if we are split that further, so it might be something we need to change again to add the split

File: examples/src/main/r/ml/bisectingKmeans.R (hunk 2 above)
Comment (felixcheung) [inline examples/src/main/r/ml/bisectingKmeans.R]: ok :) so we could add a print.summary.bisectingKMeansModel like other models :)

File: R/pkg/R/mllib_tree.R (hunk 5 above)
Comment (felixcheung) [inline R/pkg/R/mllib_tree.R]: great!

File: examples/src/main/r/ml/glm.R (hunk 3 above)
Comment (wangmiao1981) [inline examples/src/main/r/ml/glm.R]: Since this is the example not the vignettes. We can use datasets in the data/mllib directory.

File: examples/src/main/r/ml/bisectingKmeans.R (hunk 2 above)
Comment (wangmiao1981) [inline examples/src/main/r/ml/bisectingKmeans.R]: Will do in follow-up PR.

File: examples/src/main/r/ml/kmeans.R [hunk 6]
```diff
@@ -26,10 +26,12 @@ sparkR.session(appName = "SparkR-ML-kmeans-example")
 
 # $example on$
 # Fit a k-means model with spark.kmeans
-irisDF <- suppressWarnings(createDataFrame(iris))
-kmeansDF <- irisDF
-kmeansTestDF <- irisDF
-kmeansModel <- spark.kmeans(kmeansDF, ~ Sepal_Length + Sepal_Width + Petal_Length + Petal_Width,
+t <- as.data.frame(Titanic)
```
Comment (wangmiao1981) [inline examples/src/main/r/ml/kmeans.R]: kmeans_data.txt and sample_kmeans_data.txt have fewer data points than Titanic. So in this case, I am still using the Titanic dataset.

File: R/pkg/vignettes/sparkr-vignettes.Rmd [hunk 7]
```diff
@@ -565,11 +565,10 @@ We use a simple example to demonstrate `spark.logit` usage. In general, there ar
...
 Binomial logistic regression
-```{r, warning=FALSE}
-df <- createDataFrame(iris)
-# Create a DataFrame containing two classes
-training <- df[df$Species %in% c("versicolor", "virginica"), ]
-model <- spark.logit(training, Species ~ ., regParam = 0.00042)
+```{r}
+t <- as.data.frame(Titanic)
+training <- createDataFrame(t)
+model <- spark.logit(training, Sex ~ ., regParam = 0.05805813)
```
Comment (wangmiao1981) [inline R/pkg/vignettes/sparkr-vignettes.Rmd]: cv.glmnet tested.

File: examples/src/main/r/ml/glm.R [hunk 8]
```diff
@@ -25,12 +25,12 @@ library(SparkR)
...
-t <- as.data.frame(Titanic)
-training <- createDataFrame(t)
+training <- read.df("data/mllib/sample_multiclass_classification_data.txt", source = "libsvm")
 # Fit a generalized linear model of family "gaussian" with spark.glm
-gaussianDF <- training
-gaussianTestDF <- training
-gaussianGLM <- spark.glm(gaussianDF, Freq ~ Sex + Age, family = "gaussian")
+set.seed(2)
+gaussianDF <- sample(training, TRUE, 0.7)
+gaussianTestDF <- sample(training, TRUE, 0.3)
```
Comment (felixcheung) [inline examples/src/main/r/ml/glm.R]: `sample` could duplicate the same row in both training and test set.
I think we should use randomSplit instead.

**Review by felixcheung — CHANGES_REQUESTED** [review]

File: examples/src/main/r/ml/glm.R (hunk 8 above)
Comment (wangmiao1981) [inline examples/src/main/r/ml/glm.R]: OK. I will change it. Thanks!

**Comment by wangmiao1981** [comment]
@felixcheung I have made the changes per our review discussion. Thanks!

File: R/pkg/R/mllib_tree.R (hunk 5 above)
Comment (felixcheung) [inline R/pkg/R/mllib_tree.R]: Right - there are still a few examples with `Sex ~` - do you think we should change them too?

Comment (wangmiao1981) [inline R/pkg/R/mllib_tree.R]: I will change them all to survived. Thanks!

**Comment by felixcheung** [comment]
merged to master.
